doc/source/train/gbdt.rst

```diff
@@ -33,8 +33,34 @@ training parameters are passed as the ``params`` dictionary.
 
 Ray-specific params are passed in through the trainer constructors.
 
+Saving and Loading XGBoost and LightGBM Checkpoints
+---------------------------------------------------
+
+When a new tree is trained on every boosting round,
+it's possible to save a checkpoint to snapshot the training progress so far.
+:class:`~ray.train.xgboost.XGBoostTrainer` and :class:`~ray.train.lightgbm.LightGBMTrainer`
+both implement checkpointing out of the box.
+
+The only required change is to configure :class:`~ray.air.CheckpointConfig` to set
+the checkpointing frequency. For example, the following configuration will
+save a checkpoint on every boosting round and will only keep the latest checkpoint:
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __checkpoint_config_ckpt_freq_start__
+    :end-before: __checkpoint_config_ckpt_freq_end__
+
+.. tip::
+
+    Once checkpointing is enabled, you can follow :ref:`this guide <train-fault-tolerance>`
+    to enable fault tolerance.
```

Comment:
Probably don't want to link this since this is in the Deep Learning guide.

Reply:
Yeah, I was trying to share some content between the two. I linked it to a section that doesn't really depend on DL trainers. I think it's okay for now, and we can rethink the docs/user guide structure as part of the docs side of the layering project. wdyt?